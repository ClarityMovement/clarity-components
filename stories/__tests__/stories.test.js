/* eslint-env jest */
import initStoryshots from '@storybook/addon-storyshots';

jest.mock('material-ui/LinearProgress');
jest.mock('react-avatar-editor');
jest.mock('react-image-lightbox');

initStoryshots();
