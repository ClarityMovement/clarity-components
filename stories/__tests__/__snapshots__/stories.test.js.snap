// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots RenderCounter RenderCounter 1`] = `
<div
  className="root"
>
  <div>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginTop": "50px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            RenderCounter
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            RenderCounter
          </h2>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          blockOnRerender: true
          <div
            style={
              Object {
                "backgroundColor": "#f95",
                "borderRadius": "8px",
                "color": "white",
                "display": "inline-block",
                "float": "right",
                "fontSize": "8pt",
                "fontWeight": "bold",
                "padding": "1px 5px",
                "textAlign": "center",
                "top": "2px",
              }
            }
          >
            1
          </div>
        </div>
      </div>
    </div>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginTop": "50px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            NOTICE: You may see some extra times.
          </p>
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            Thatâ€™s a bug of storybook happens when you enter this page directly.
          </p>
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            You can click another menu and re-enter this route to see real render times.
          </p>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            style={
              Object {
                "backgroundColor": "#fafafa",
                "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
                "fontSize": ".88em",
                "lineHeight": 1.5,
                "overflowX": "scroll",
                "padding": ".5rem",
              }
            }
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  div
                </span>
                <span />
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &gt;
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 48,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  blockOnRerender: true
                </span>
              </div>
              <div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;
                    DivRefreshable
                  </span>
                  <span />
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 48,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;
                    RenderCounter
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        onRerender
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          <span>
                            <span
                              style={
                                Object {
                                  "color": "#170",
                                }
                              }
                            >
                              function handler(props, next) {return function (count) {return log('handler blocked', props, count);};}()
                            </span>
                          </span>
                        </span>
                      </span>
                       
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    /&gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;/
                    DivRefreshable
                    &gt;
                  </span>
                </div>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;/
                  div
                  &gt;
                </span>
              </div>
            </div>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              RenderCounter
              " Component
            </h2>
            <table
              style={
                Object {
                  "borderCollapse": "separate",
                  "borderSpacing": "10px 5px",
                  "marginLeft": -10,
                }
              }
            >
              <thead>
                <tr>
                  <th>
                    property
                  </th>
                  <th>
                    propType
                  </th>
                  <th>
                    required
                  </th>
                  <th>
                    default
                  </th>
                  <th>
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    initialCount
                  </td>
                  <td>
                    number
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        1
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    onRerender
                  </td>
                  <td>
                    func
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    -
                  </td>
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots RenderCounter StatelessRenderCounter 1`] = `
<div
  className="root"
>
  <div>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginTop": "50px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            RenderCounter
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            StatelessRenderCounter
          </h2>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div>
          blockOnRerender: true
          <div
            style={
              Object {
                "backgroundColor": "#f95",
                "borderRadius": "8px",
                "color": "white",
                "display": "inline-block",
                "float": "right",
                "fontSize": "8pt",
                "fontWeight": "bold",
                "padding": "1px 5px",
                "textAlign": "center",
                "top": "2px",
              }
            }
          >
            1
          </div>
        </div>
      </div>
    </div>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "
              -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
              \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
            ",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginTop": "50px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            NOTICE: You may see some extra times.
          </p>
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            Thatâ€™s a bug of storybook happens when you enter this page directly.
          </p>
          <p
            style={
              Object {
                "WebkitFontSmoothing": "antialiased",
                "color": "#444",
                "fontFamily": "
                  -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Roboto\\",
                  \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif
                ",
                "fontSize": "15px",
              }
            }
          >
            You can click another menu and re-enter this route to see real render times.
          </p>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            style={
              Object {
                "backgroundColor": "#fafafa",
                "fontFamily": "Menlo, Monaco, \\"Courier New\\", monospace",
                "fontSize": ".88em",
                "lineHeight": 1.5,
                "overflowX": "scroll",
                "padding": ".5rem",
              }
            }
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  div
                </span>
                <span />
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &gt;
                </span>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 48,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  blockOnRerender: true
                </span>
              </div>
              <div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;
                    DivRefreshable
                  </span>
                  <span />
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 48,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;
                    withRenderCount(Counter)
                  </span>
                  <span>
                    <span>
                       
                      <span
                        style={Object {}}
                      >
                        onRerender
                      </span>
                      <span>
                        =
                        <span
                          style={Object {}}
                        >
                          <span>
                            <span
                              style={
                                Object {
                                  "color": "#170",
                                }
                              }
                            >
                              function handler(props, next) {return function (count) {return log('handler blocked', props, count);};}()
                            </span>
                          </span>
                        </span>
                      </span>
                       
                    </span>
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    /&gt;
                  </span>
                </div>
                <div
                  style={
                    Object {
                      "paddingLeft": 33,
                      "paddingRight": 3,
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "color": "#777",
                      }
                    }
                  >
                    &lt;/
                    DivRefreshable
                    &gt;
                  </span>
                </div>
              </div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;/
                  div
                  &gt;
                </span>
              </div>
            </div>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              Counter
              " Component
            </h2>
            <table
              style={
                Object {
                  "borderCollapse": "separate",
                  "borderSpacing": "10px 5px",
                  "marginLeft": -10,
                }
              }
            >
              <thead>
                <tr>
                  <th>
                    property
                  </th>
                  <th>
                    propType
                  </th>
                  <th>
                    required
                  </th>
                  <th>
                    default
                  </th>
                  <th>
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    count
                  </td>
                  <td>
                    number
                  </td>
                  <td>
                    yes
                  </td>
                  <td>
                    -
                  </td>
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              withRenderCount(Counter)
              " Component
            </h2>
            <table
              style={
                Object {
                  "borderCollapse": "separate",
                  "borderSpacing": "10px 5px",
                  "marginLeft": -10,
                }
              }
            >
              <thead>
                <tr>
                  <th>
                    property
                  </th>
                  <th>
                    propType
                  </th>
                  <th>
                    required
                  </th>
                  <th>
                    default
                  </th>
                  <th>
                    description
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    initialCount
                  </td>
                  <td>
                    number
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    <span>
                      <span
                        style={
                          Object {
                            "color": "#a11",
                          }
                        }
                      >
                        1
                      </span>
                    </span>
                  </td>
                  <td />
                </tr>
                <tr>
                  <td>
                    onRerender
                  </td>
                  <td>
                    func
                  </td>
                  <td>
                    no
                  </td>
                  <td>
                    -
                  </td>
                  <td />
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Welcome to Storybook 1`] = `
<div
  style={
    Object {
      "fontFamily": "\\"Helvetica Neue\\", Helvetica, \\"Segoe UI\\", Arial, freesans, sans-serif",
      "lineHeight": 1.4,
      "margin": 15,
    }
  }
>
  <h1>
    Welcome to STORYBOOK
  </h1>
  <p>
    This is a UI component dev environment for your app.
  </p>
  <p>
    We've
     added some basic stories inside the 
    <code
      style={
        Object {
          "backgroundColor": "#f3f2f2",
          "border": "1px solid #eae9e9",
          "borderRadius": 4,
          "color": "#3a3a3a",
          "fontSize": 15,
          "fontWeight": 600,
          "padding": "2px 5px",
        }
      }
    >
      src/stories
    </code>
     directory.
    <br />
    A story is a single state of one or more UI components. You can have as many stories as you want.
    <br />
    (Basically a story is like a visual test case.)
  </p>
  <p>
    See these sample 
    <a
      href="#/"
      onClick={[Function]}
      style={
        Object {
          "borderBottom": "1px solid #1474f3",
          "color": "#1474f3",
          "paddingBottom": 2,
          "textDecoration": "none",
        }
      }
    >
      stories
    </a>
     for a component called 
    <code
      style={
        Object {
          "backgroundColor": "#f3f2f2",
          "border": "1px solid #eae9e9",
          "borderRadius": 4,
          "color": "#3a3a3a",
          "fontSize": 15,
          "fontWeight": 600,
          "padding": "2px 5px",
        }
      }
    >
      RenderCounter
    </code>
    .
  </p>
  <p>
    Just like that, you can add your own components as stories.
    <br />
    You can also edit those components and see changes right away.
    <br />
    (Try editing the 
    <code
      style={
        Object {
          "backgroundColor": "#f3f2f2",
          "border": "1px solid #eae9e9",
          "borderRadius": 4,
          "color": "#3a3a3a",
          "fontSize": 15,
          "fontWeight": 600,
          "padding": "2px 5px",
        }
      }
    >
      Button
    </code>
     component located at 
    <code
      style={
        Object {
          "backgroundColor": "#f3f2f2",
          "border": "1px solid #eae9e9",
          "borderRadius": 4,
          "color": "#3a3a3a",
          "fontSize": 15,
          "fontWeight": 600,
          "padding": "2px 5px",
        }
      }
    >
      src/stories/Button.js
    </code>
    .)
  </p>
  <p>
    This is just one thing you can do with Storybook.
    <br />
    Have a look at the 
    <a
      href="https://github.com/kadirahq/react-storybook"
      rel="noopener noreferrer"
      style={
        Object {
          "borderBottom": "1px solid #1474f3",
          "color": "#1474f3",
          "paddingBottom": 2,
          "textDecoration": "none",
        }
      }
      target="_blank"
    >
      React Storybook
    </a>
     repo for more information.
  </p>
  <p
    style={
      Object {
        "opacity": 0.5,
      }
    }
  >
    <b>
      NOTE:
    </b>
    <br />
    Have a look at the 
    <code
      style={
        Object {
          "backgroundColor": "#f3f2f2",
          "border": "1px solid #eae9e9",
          "borderRadius": 4,
          "color": "#3a3a3a",
          "fontSize": 15,
          "fontWeight": 600,
          "padding": "2px 5px",
        }
      }
    >
      .storybook/webpack.config.js
    </code>
     to add webpack loaders and plugins you are using in this project.
  </p>
</div>
`;

exports[`Storyshots Welcome to react-render-counter 1`] = `
<div>
  <h1
    className={undefined}
    data-sourcepos={undefined}
  >
    hhh
  </h1>
  <h2
    className={undefined}
    data-sourcepos={undefined}
  >
    h2
  </h2>
</div>
`;

exports[`Storyshots Welcome to stupid-rc-starter 1`] = `
<div>
  <h1
    className={undefined}
    data-sourcepos={undefined}
  >
    hhh
  </h1>
  <h2
    className={undefined}
    data-sourcepos={undefined}
  >
    h2
  </h2>
</div>
`;
